BLSQ/iaso
# IA-1247 duplicate source version number

Add a constraint that a Data soucre cannot have two version with the same number.

I checked all the installation in production and there was only one case in the TRASH data source on prod I fixed it.

Note that when applying the migration locally it might refuse if you have duplicate. 
Before applying the migration, go to the Source Version page in the admin and check if you have duplicate number for the same source if it is the case change them.

The following SQL query was used to check the presence of dupliate
```sql
select  (select name from iaso_datasource where iaso_datasource.id=data_source_id),
           data_source_id, number, count(*)
from iaso_sourceversion group by data_source_id, number order by 4 desc, 
```
To only have duplicate
```sql
select  (select name from iaso_datasource where iaso_datasource.id=data_source_id), data_source_id, number, count(*) from iaso_sourceversion group by data_source_id, number order by 4 desc, 1 having count(*) > 1
```

You can execute the query in the sql dashboard in the web interface
`/explore/?sql=select++%28select+name+from+iaso_datasource+where+iaso_datasource.id%3Ddata_source_id%29%2C+data_source_id%2C+number%2C+count%28%2A%29+from+iaso_sourceversion+group+by+data_source_id%2C+number+order+by+4+desc%2C+1+--+having+count%28%2A%29+%3E+1%3AtSe8bFFAXmhUaU2WwvbFYJJhqHR22DqFFPX-oFtyPzI`


## Self proof reading checklist


## Changes

- Add constraint and corresponding migration
- Also fix a crash in the frontend for the task  page

## How to test
From original bug report 
reproduce:

1.     go to /api/sourceversions
2.     use interface to create a new version with the number of your choice
3.     click post
4.     click post again

expected
-     new version is created on first click
-     creation is prevented on second click because a version with that number already exists

actual
-    2 versions with the same number are created
